packages/experimental/Shadow/src/Shadow.tsx

```diff
@@ -22,12 +21,95 @@ export const Shadow = stagedComponent((props: ShadowProps) => {
       }
     }
 
-    const childWithKeyShadow = React.cloneElement(child, mergeProps(child.props, { style: shadowTokenStyleSet.key }));
+    const { style: childStyle = {}, ...restOfChildProps } = child.props;
+
+    const shadowViewStyleProps = getStylePropsForShadowViews(childStyle, shadowTokenStyleSet);
+    const innerShadowViewProps = mergeProps(restOfChildProps, shadowViewStyleProps.inner);
+    const outerShadowViewProps = mergeProps(final, shadowViewStyleProps.outer);
+
+    const childWithInnerShadow = React.cloneElement(child, innerShadowViewProps);
 
-    return <View {...mergedProps}>{childWithKeyShadow}</View>;
+    return <View {...outerShadowViewProps}>{childWithInnerShadow}</View>;
   };
 });
 
+const getStylePropsForShadowViews = memoize(getStylePropsForShadowViewsWorker);
+function getStylePropsForShadowViewsWorker(childStyleProps: any, shadowTokenStyleSet: any) {
+  const {
+    borderBottomWidth,
+    borderEndWidth,
+    borderLeftWidth,
+    borderRightWidth,
+    borderStartWidth,
+    borderTopWidth,
+    borderWidth,
+
+    margin,
+    marginBottom,
+    marginEnd,
+    marginHorizontal,
+    marginLeft,
+    marginRight,
+    marginStart,
+    marginTop,
+    marginVertical,
+
+    padding,
+    paddingBottom,
+    paddingEnd,
+    paddingHorizontal,
+    paddingLeft,
+    paddingRight,
+    paddingStart,
+    paddingTop,
+    paddingVertical,
+
+    ...restOfChildStyleProps
+  } = childStyleProps;
+
+  return {
+    inner: {
+      style: {
+        backgroundColor: 'red', // will not be shown, just needed in macOS/iOS to ensure buggy behaviour is suppressed
```

Comment:
I think this should be a copy of the outer view's background color, for "bleeding through" reasons discussed already.